CRM/Utils/Crypt.php

```diff
@@ -51,20 +51,13 @@ public static function encrypt($string) {
     if (empty($string)) {
       return $string;
     }
-
-    if (function_exists('mcrypt_module_open') &&
-      defined('CIVICRM_SITE_KEY')
-    ) {
-      $td = mcrypt_module_open(MCRYPT_RIJNDAEL_256, '', MCRYPT_MODE_ECB, '');
-      // ECB mode - iv not needed - CRM-8198
-      $iv = '00000000000000000000000000000000';
-      $ks = mcrypt_enc_get_key_size($td);
-      $key = substr(sha1(CIVICRM_SITE_KEY), 0, $ks);
-
-      mcrypt_generic_init($td, $key, $iv);
-      $string = mcrypt_generic($td, $string);
-      mcrypt_generic_deinit($td);
-      mcrypt_module_close($td);
+    if (defined('CIVICRM_SITE_KEY') && function_exists('openssl_encrypt')) {
+      $raw = defined('OPENSSL_RAW_DATA') ? OPENSSL_RAW_DATA : TRUE;
+      $key = openssl_digest(CIVICRM_SITE_KEY, 'sha256');
+      $method = 'AES-256-ECB';
+      $ivSize = 0;
+      $iv = substr($key, 0, $ivSize);
+      $string = openssl_encrypt($string, $method, $key, $raw, $iv);
```

Comment:
I was aiming as a drop in replacement and i believed that this was the closest to the mcrypt which i thought was the case, i would prefer a drop in replacement and see this as being part of it, I think b is good but going to be crazy to solve